daprdocs/content/en/operations/components/pluggable-components/pluggable-components-discovery.md

```diff
@@ -0,0 +1,229 @@
+---
+type: docs
+title: "How-To: Discover a pluggable component"
+linkTitle: "How To: Discover a pluggable component"
+weight: 4500
+description: "Learn how to help Dapr discover your pluggable component"
+---
+
+[uds]: https://en.wikipedia.org/wiki/Unix_domain_socket
+
+## Service Discovery Process
+
+Pluggable, [gRPC-based](https://grpc.io/) components are typically run as containers or processes that need to communicate with the Dapr main process via [Unix Domain Sockets][uds]. They are automatically discovered and registered in runtime by Dapr using the following steps:
+
+1. The Component listens to an [Unix Domain Socket][uds] placed on the shared volume.
+2. The Dapr runtime lists all [Unix Domain Socket][uds] in the shared volume.
+3. The Dapr runtime connects with the socket and uses gRPC reflection to discover all services that such component implements.
+
+A single component can implement multiple [building blocks]({{< ref building-blocks-concept.md >}}) at once.
+
+<img src="/images/grpc-components.png" width=50%>
+
+While Dapr's built-in components come [ready to be used out of the box](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md), pluggable components require a few setup steps before they can be used with Dapr.
+
+1. Pluggable components need to be started and ready to take requests _before_ Dapr itself is started.
+2. The [Unix Domain Socket][uds] file used used for the pluggable component communication need to be made accessible to both Dapr and pluggable component.
+
+Dapr does not interfere with orchestrating components containers creation and deployment. This is your domain, and it will be different depending on how Dapr and your components are run:
+
+- In standalone mode, as processes or containers, or
+- In Kubernetes, as containers.
+
+This will also change the mechanisms available to share [Unix Domain Socket][uds] files between Dapr and pluggable components.
+
+{{% alert title="Note" color="primary" %}}
+As a prerequisite the running operating system must supports Unix Domain Sockets, any UNIX or UNIX-like system (Mac, Linux, or for local development [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) for Windows users) should be sufficient.
+{{% /alert %}}
+
+Select your running environment to begin making your component discoverable by Dapr.
+
+{{< tabs "Standalone" "Kubernetes" >}}
+
+{{% codetab %}}
```

Comment:
I would say that codetabs are great when you can choose two journeys that are completely independent. I don't think this is really the case here. Yes, there is content duplication, but we provide more details on the Standalone path and just rush through on the Kubernetes one. For instance, how we come up with the `spec.type` field in the component spec is explained better in the Standalone route. The default location for UDS is not even mentioned in the k8s path, just that you better share it.

I would rather remove these codetabs _or_ have some of these common topics defined/detailed before so that concepts are presented once and presented right.

Thoughts?

Reply:
Yeah, but if we are having these tabs, than we better ensure content on both is solid. Right not the k8s one could use some love.